### PR TITLE
Core, Webhost, Docs: Replace all usages of player settings

### PR DESCRIPTION
**Pull request by nicholassaylor**
## What is this fixing or adding?
1. Replaces the remaining usages of player settings with player options
    * Made sure to avoid replacing any instances of the settings API
    * Most replacements are in docs that weren't changed in #2622
    * No instances of `/player-options` remain

2. Changes Generate.py to reference player options, changing the `--samesettings` argument to `--sameoptions`

3. Removed backwards compatability for player-settings link
    * `/player-settings` links should no longer auto-redirect to `/player-options` page

## How was this tested?
1. Searched for all instances of "settings" or "player-settings" and replaced any that did not refer to the Settings API
2. Looked at rendered markdown
3. Ran Generate.py with `--sameoptions` argument and got expected output
    * Would appreciate if another person could test argument change to make sure that I did everything correct (first time touching core)

### CODE REVIEW

**Comment by Exempt-Medic** [comment]
There's also a line in here you'll want to change: https://github.com/ArchipelagoMW/Archipelago/blob/main/worlds/archipidle/docs/guide_en.md

**Comment by nicholassaylor** [comment]
> There's also a line in here you'll want to change: https://github.com/ArchipelagoMW/Archipelago/blob/main/worlds/archipidle/docs/guide_en.md Specifically "setting page above"

Thank you, it took me a few reads to see where it was

**Comment by Exempt-Medic** [comment]
You can probably also update these:
* https://github.com/ArchipelagoMW/Archipelago/blob/main/WebHostLib/misc.py#L40-L43
* https://github.com/ArchipelagoMW/Archipelago/blob/main/WebHostLib/misc.py#L52-L55

**Comment by nicholassaylor** [comment]
> You can probably also update these:
> 
>     * https://github.com/ArchipelagoMW/Archipelago/blob/main/WebHostLib/misc.py#L40-L43
> 
>     * https://github.com/ArchipelagoMW/Archipelago/blob/main/WebHostLib/misc.py#L52-L55
>       Hmm, actually, looking at instances of `weighted-settings` that one probably can't be changed yet

Ahh, I didn't notice that there was specific backwards compat with weighted right now. I'll look into and see what I can find out. This PR can be marked as `waiting on: author` for now until I figure out the WebHost-specific changes I would need

**Comment by Exempt-Medic** [comment]
> Ahh, I didn't notice that there was specific backwards compat with weighted right now. I'll look into and see what I can find out. This PR can be marked as `waiting on: author` for now until I figure out the WebHost-specific changes I would need

I think it'd be fine if this PR just did everything for `player-settings`. The weighted page should probably be left for later anyway

**Comment by nicholassaylor** [comment]
> I think it'd be fine if this PR just did everything for `player-settings`. The weighted page should probably be left for later anyway
I suppose that works, especially with the revamped options pages in the works.

**Comment by ScipioWright** [comment]
I see newlines were added in several of these.
I know for option descriptions, newlines are taken into account and, in some cases, make it look weird on the website.
Did you check to make sure that doesn't happen with these pages too?

**Comment by nicholassaylor** [comment]
I can double check, all of those newline were added in by PyCharm automatically

**Comment by alwaysintreble** [comment]
unsure if it's in scope for this PR but there are some remaining f strings in Generate.py that still reference player settings

**Comment by LegendaryLinux** [comment]
Thus far this does not conflict with the option-gruops PR, so it has my approval pending any further changes.

**Comment by nicholassaylor** [comment]
> unsure if it's in scope for this PR but there are some remaining f strings in Generate.py that still reference player settings

I did my best to address the Generate.py f strings and arguments. I think I caught everything in there



> I see newlines were added in several of these. I know for option descriptions, newlines are taken into account and, in some cases, make it look weird on the website. Did you check to make sure that doesn't happen with these pages too?

I reverted all of the changes that were just newlines put in automatically. None of the files should have different newline behaviors.

File: worlds/archipidle/docs/guide_fr.md [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
 ## Rejoindre une partie MultiWorld
-1. Générez un fichier `.yaml` à partir de la [page des paramètres du lecteur ArchipIDLE](/games/ArchipIDLE/player-settings)
+1. Générez un fichier `.yaml` à partir de
+   la [page des paramètres du lecteur ArchipIDLE](/games/ArchipIDLE/player-options)
 2. Ouvrez le client ArchipIDLE dans votre navigateur Web en :
-     - Accédez au [Client ArchipIDLE](http://idle.multiworld.link)
-     - Téléchargez le client et exécutez-le localement à partir du
-       [Page des versions d'ArchipIDLE GitHub](https://github.com/ArchipelagoMW/archipidle/releases)
+    - Accédez au [Client ArchipIDLE](http://idle.multiworld.link)
+    - Téléchargez le client et exécutez-le localement à partir du [Page des versions d'ArchipIDLE GitHub](https://github.com/ArchipelagoMW/archipidle/releases)
```
Comment (Exempt-Medic) [inline worlds/archipidle/docs/guide_fr.md:8]: You split the previous line into two but not this one. I'm just confused as to why

**Review by Exempt-Medic — CHANGES_REQUESTED** [review]
Two small things that stuck out. Everything else looks good

File: worlds/minecraft/docs/minecraft_sv.md [hunk 2]
```diff
@@ -103,8 +103,6 @@ shuffle_structures:
   off: 0
 ```
 
-För mer detaljer om vad varje inställning gör, kolla standardinställningen `PlayerSettings.yaml` som kommer med
```
[the remark this answers is not in the document]
Comment (nicholassaylor) [reply worlds/minecraft/docs/minecraft_sv.md:106]: This paragraph references the PlayerSettings.yaml file that no longer exists as of #3062 . The document overall is very outdated, but I don't know any Swedish to update it.

[the remark this answers is not in the document]
Comment (Exempt-Medic) [reply worlds/minecraft/docs/minecraft_sv.md:106]: Sounds good to me then!

File: worlds/archipidle/docs/guide_fr.md (hunk 1 above)
Comment (nicholassaylor) [reply worlds/archipidle/docs/guide_fr.md:7]: I just missed the first line split when going back and trying to correct PyCharm's forced formatting, it should be fixed now

**Review by Exempt-Medic — APPROVED** [review]
All the changes LGTM. Searched for other instances of "settings" in the options sense and didn't find any

File: WebHostLib/misc.py [hunk 3]
```diff
@@ -49,12 +49,6 @@ def weighted_options():
     return render_template("weighted-options.html")
 
 
-# TODO for back compat. remove around 0.4.5
```
Comment (LegendaryLinux) [inline WebHostLib/misc.py:52]: Removing this route is out of scope for this PR as a terminology update.

**Review by LegendaryLinux — APPROVED** [review]
